File: FiveRockingFingers/FRF.Core/Models/ArtifactRelation.cs
namespace FRF.Core.Models
{
    public class ArtifactRelation
    {
        public int Artifact1Id { get; set; }
        public Artifact Artifact1 { get; set; }

        public int Artifact2Id { get; set; }
        public Artifact Artifact2 { get; set; }

        public string Artifact1Property { get; set; }
        public string Artifact2Property { get; set; }
        public int RelationTypeId { get; set; }
    }
}

File: FiveRockingFingers/FRF.Web/ClientApp/src/interfaces/ArtifactRelation.ts
import Artifact from './Artifact';
import AwsArtifact from './AwsArtifact';
import KeyValueStringPair from './KeyValueStringPair';

export default interface ArtifactRelation {
    id: string | null;
    artifact1: Artifact;
    artifact2: Artifact;
    setting1: KeyValueStringPair;
    setting2: KeyValueStringPair;
    relationTypeId: number;
}